src/code128.ps.src

```diff
@@ -396,6 +398,171 @@ begin
 
     encoding (new) eq {
 
+        % Include pseudo characters for GS1-128 Composite linkage identifiers
+        /seta <<fn3 96 fn2 97 fn1 102 stp 106 lka 100 lkc  99>> def
+        /setb <<fn3 96 fn2 97 fn1 102 stp 106 lka  99 lkc 101>> def
+        /setc <<              fn1 102 stp 106 lka 101 lkc 100>> def
+
+        /text msglen string def
+        0 1 msglen 1 sub {
+            /i exch def
+            text i msg i get dup 0 lt { pop 32 } if put
+        } for
+
+        65 dict begin
+        % Encoder states are named A0, B0, C0, A1, B1, and C1.
+        % A, B, and C are the Code 128 code sets.
+        % 0 is for ASCII and 1 is for extended ASCII range.
+        % The encoder relies on this order a lot, but it uses
+        % the "state_priority" list for prioritizing states.
+        % Strings are a convenient way of defining latch
+        % sequences. The strings map to Code 128 instructions.
+        % The (dddc) sequences are never inserted, but they
+        % are in the lists to provide the correct length:
+        %                      Prior state
+        %          A0    B0    C0     A1    B1    C1
+        /latch_a0 [()    (e)   (e)    (ee)  (eee) (eee) ] def
+        /latch_b0 [(d)   ()    (d)    (ddd) (dd)  (ddd) ] def
+        /latch_c0 [(c)   (c)   ()     (eec) (ddc) (dddc)] def
+        /latch_a1 [(ee)  (eee) (eee)  ()    (e)   (e)   ] def
+        /latch_b1 [(ddd) (dd)  (ddd)  (d)   ()    (d)   ] def
+        /latch_c1 [(eec) (ddc) (dddc) (c)   (c)   ()    ] def
+
+        /latch_length_a0 [latch_a0 {length} forall] def
+        /latch_length_a1 [latch_a1 {length} forall] def
+        /latch_length_b0 [latch_b0 {length} forall] def
+        /latch_length_b1 [latch_b1 {length} forall] def
+        /latch_length_c0 [latch_c0 {length} forall] def
+        /latch_length_c1 [latch_c1 {length} forall] def
+
+        % Backtracking needs a way of mapping states to sequences:
```

Comment:
Nit: I prefer to now have comments ending with `:`.

Reply:
Gocha.

Reply:
Typo in what I wrote: s/now/not/

My bad.